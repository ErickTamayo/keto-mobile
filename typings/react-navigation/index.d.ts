/* eslint-disable */
import ReactNavigation from 'react-navigation'

declare module 'react-navigation' {
  export interface StackNavigatorConfig {
    cardShadowEnabled: boolean
  }
}
