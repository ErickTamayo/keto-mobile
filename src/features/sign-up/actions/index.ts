import { SELECT_GENDER } from '../constants'

export interface SelectGender {
  type: SELECT_GENDER
  gender: 'male' | 'female' | null
}

export const selectGender = (gender: SelectGender['gender']): SelectGender => ({
  type: SELECT_GENDER,
  gender,
})
