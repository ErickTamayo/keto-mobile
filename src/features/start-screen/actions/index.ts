import { ADD_COUNTER, USER_LOGOUT } from '../constants'

export interface AddCounter {
  type: ADD_COUNTER
}

export interface UserLogout {
  type: USER_LOGOUT
}

export const addCounter = (): AddCounter => ({
  type: ADD_COUNTER,
})

export const userLogout = (): UserLogout => ({
  type: USER_LOGOUT,
})
