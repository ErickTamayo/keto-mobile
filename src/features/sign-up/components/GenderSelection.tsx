import * as React from 'react'
import { Text, View } from 'react-native'
import { NavigationInjectedProps } from 'react-navigation'
import { StateProps, DispatchProps } from '../containers/GenderSelection'
import st from '../../../styles'
import CardWithHeaderAndButton from '../../../components/card-with-header-and-button/CardWithHeaderAndButton'

export interface Props extends StateProps, DispatchProps, NavigationInjectedProps {}

export default class GenderSelection extends React.Component<Props, object> {
  private navigateToOtherParameters = () => {
    const { navigation } = this.props
    navigation.navigate('OtherParameters')
  }

  // private navigateToLogIn = () => this.props.navigation.navigate('Login')

  public render(): JSX.Element {
    // const { counter, addCounter } = this.props
    const disabled = false
    return (
      <View style={[st.flex.f1, st.items.center, st.justify.center, st.bg.greyLightest]}>
        <CardWithHeaderAndButton
          title="What's your gender?"
          description="We will make sure you get better and personalized results"
          buttonText="Continue"
          onButtonPress={this.navigateToOtherParameters}
          disabled={disabled}
        >
          <Text>Radio Buttons here</Text>
          <Text>Radio Buttons here</Text>
          <Text>Radio Buttons here</Text>
          <Text>Radio Buttons here</Text>
          <Text>Radio Buttons here</Text>
          <Text>Radio Buttons here</Text>
          {/* <RadioButton variant="lg" checked={true} value="female" onChange={this.handleChange}>
            Female
          </RadioButton>
          <RadioButton variant="lg" checked={false} value="male" onChange={this.handleChange}>
            Male
          </RadioButton> */}
        </CardWithHeaderAndButton>
      </View>
    )
  }
}
