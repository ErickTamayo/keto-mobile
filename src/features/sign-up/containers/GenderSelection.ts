import { connect } from 'react-redux'
import { Dispatch } from 'redux'
import { State } from '../records/SignUpState'
import GenderSelection from '../components/GenderSelection'
import { selectGender, SelectGender } from '../actions'

export interface StateProps {
  gender: 'male' | 'female' | null
}

export interface DispatchProps {
  selectGender: (gender: 'male' | 'female') => SelectGender
}

export const mapStateToProps = ({ signUp }: { signUp: State }): StateProps => {
  return {
    gender: signUp.gender,
  }
}

export const mapDispatchToProps = (dispatch: Dispatch): DispatchProps => ({
  selectGender: (gender: 'male' | 'female'): SelectGender => dispatch(selectGender(gender)),
})

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(GenderSelection)
