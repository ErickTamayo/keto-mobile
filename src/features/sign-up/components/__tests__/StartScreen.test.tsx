import 'react-native'
import * as React from 'react'
import { shallow } from 'enzyme'
import { NavigationScreenProp, NavigationRoute, NavigationParams } from 'react-navigation'
import GenderSelection, { Props } from '../GenderSelection'
import { SelectGender } from '../../actions'

describe('<GenderSelection />', () => {
  it('renders correctly', () => {
    const props: Props = {
      gender: 'female',
      selectGender: () => ({} as SelectGender),
      navigation: {} as NavigationScreenProp<NavigationRoute<NavigationParams>, NavigationParams>,
    }
    const wrapper = shallow(<GenderSelection {...props} />)
    expect(wrapper).toMatchSnapshot()
  })
})
