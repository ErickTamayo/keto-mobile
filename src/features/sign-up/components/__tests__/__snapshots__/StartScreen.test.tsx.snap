// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StartScreen /> renders correctly 1`] = `
<Component
  style={
    Array [
      Object {
        "flex": 1,
      },
      Object {
        "alignItems": "center",
      },
      Object {
        "justifyContent": "center",
      },
      Object {
        "backgroundColor": "#f7f7f7",
      },
    ]
  }
>
  <CardWithHeaderAndButton
    buttonText="Continue"
    description="We will make sure you get better and personalized results"
    disabled={false}
    onButtonPress={[Function]}
    title="What's your gender?"
  >
    <Component>
      Radio Buttons here
    </Component>
    <Component>
      Radio Buttons here
    </Component>
    <Component>
      Radio Buttons here
    </Component>
    <Component>
      Radio Buttons here
    </Component>
    <Component>
      Radio Buttons here
    </Component>
    <Component>
      Radio Buttons here
    </Component>
  </CardWithHeaderAndButton>
</Component>
`;
